### PR TITLE
Update travis.yml to include support for tagging

### PR DESCRIPTION
**Pull request by onyiny-ang**
Addresses #152

### CODE REVIEW

**Comment by onyiny-ang** [comment]
Test failing #159

File: scripts/imagebuild.sh [hunk 1]
```diff
@@ -23,9 +23,15 @@ dockerfile_dir="${base_dir}/images/federation-v2"
 
 [ -f "$base_dir/bin/controller-manager" ] || { echo "$base_dir/bin/controller-manager not found" ; exit 1 ;}
 
-if [[ "${TRAVIS_BRANCH}" != "master" ]]; then
-  echo "Not on master branch. Image build skipped." >&2
-  exit 0
+if [[ "${TRAVIS_TAG}" =~ ^[0-9]+\.[0-9]+\.[0-9]?\d$ ]]; then
+    echo "Pushing images with tags '${TRAVIS_TAG}' and 'latest'."
```
Comment (gyliu513) [inline scripts/imagebuild.sh]: What do you mean of `tags '${TRAVIS_TAG}' and 'latest'.`? Why including `latest` here?

Comment (shashidharatd) [reply scripts/imagebuild.sh]: we are not pushing the image with latest tag. either message needs updating or we should push image with latest tag.

File: scripts/imagebuild.sh [hunk 2]
```diff
@@ -23,9 +23,15 @@ dockerfile_dir="${base_dir}/images/federation-v2"
...
+    export TAG="${TRAVIS_TAG}"
+elif [[ "${TRAVIS_BRANCH}" == "master" ]]; then
+    echo "Pushing images with default tags (git sha and 'canary')."
```
Comment (shashidharatd) [inline scripts/imagebuild.sh]: we are not pushing image with `git sha` as a tag, so the message is misleading.

**Comment by onyiny-ang** [comment]
@shashidharatd @gyliu513 removed misleading tag messages from imagebuild.sh

**Comment by shashidharatd** [comment]
@onyiny-ang, Thanks for the update. couple of things.
1. Why do we use the tag canary instead of latest? for an end user who is familiar with docker using the image without tag is normal if he does not care about version.
2. For every merged pr, it makes sense to have 2 docker pushes, one for `git sha` and another one with `latest` tag (considering the first comment).

**Comment by onyiny-ang** [comment]
@shashidharatd 
re: canary, it's a convention: http://www.yourdictionary.com/canary-build
Though I haven't fully determined what system we want to go with, I was looking at the tagging in place for svc-catalog: https://github.com/kubernetes-incubator/service-catalog/blob/master/contrib/travis/deploy.sh
In the case of service catalog, version tag and 'latest' are pushed together as more stable and well tested versions whereas git sha and 'canary' are used on master to alert users to the fact that they don't represent a stable release. This is not what is implemented in my PR atm because I'm not sure what system we want to follow for federation.
I welcome feedback on this.
@pmorie @marun @font @irfanurrehman

File: scripts/imagebuild.sh [hunk 3]
```diff
@@ -23,9 +23,15 @@ dockerfile_dir="${base_dir}/images/federation-v2"
 
 [ -f "$base_dir/bin/controller-manager" ] || { echo "$base_dir/bin/controller-manager not found" ; exit 1 ;}
 
-if [[ "${TRAVIS_BRANCH}" != "master" ]]; then
-  echo "Not on master branch. Image build skipped." >&2
-  exit 0
+if [[ "${TRAVIS_TAG}" =~ ^[0-9]+\.[0-9]+\.[0-9]?\d$ ]]; then
+    echo "Pushing images with tag '${TRAVIS_TAG}'."
+    export TAG="${TRAVIS_TAG}"
```
Comment (pmorie) [inline scripts/imagebuild.sh]: You don't need to export this; you can just say `tag=` and then reference elsewhere as `${tag}`

**Comment by onyiny-ang** [comment]
Addresses #152

**Pull request merged**